day22/permutations.py

```diff
@@ -0,0 +1,20 @@
+def permutations(arr):
+    allPermutations = list()
+
+    if len(arr) <= 1:
+        return [arr]
+    else:
+        for i in xrange(len(arr)):
+            for permutation in permutations(arr[1:]):
+                # when i is 0, arr[0:1] is at the beginning of this list
```

Comment:
The comment is to remind me that this one line does handle all permutations, including having the first element of `arr` at the beginning, middle, and end